crates/task-impls/src/quorum_proposal_recv/handlers.rs

```diff
@@ -90,55 +98,41 @@ async fn validate_proposal_liveness<TYPES: NodeType, I: NodeImplementation<TYPES
     .await;
 
     if liveness_check {
-        let new_view = proposal.data.view_number + 1;
-
-        // This is for the case where we form a QC but have not yet seen the previous proposal ourselves
-        let should_propose = task_state.quorum_membership.leader(new_view) == task_state.public_key
-            && high_qc.view_number() == proposal.data.view_number();
-
-        if should_propose {
-            debug!(
-                "Attempting to publish proposal after voting for liveness; now in view: {}",
-                *new_view
-            );
-            broadcast_event(
-                HotShotEvent::QuorumProposalLivenessValidated(proposal.data.clone()).into(),
-                event_sender,
-            )
-            .await;
-        }
-
-        return Some(proposal.data.clone());
+        return QuorumProposalValidity::LivenessValid;
     }
 
-    None
+    QuorumProposalValidity::Failed
```

Comment:
@bfish713 @jparr721 This is the change I mentioned in the pair programming that I'd like a second pair of eyes.

With the previous logic, this should be `QuorumProposalValidity::FullyValid`, but as discussed, I think it makes more sense to mark this as `Failed`, though it probably doesn't matter logically.

Reply:
In the happy path where we have the data in the internal state but the liveness fails, we return `Failed`, so here I think we should be consistent.

I think the ultimate question is whether we should call `cancel_tasks` in `handle` in the quorum proposal recv event, or only log the error. If it's the former, we should update both the happy and sad cases to return `FullyValid` if liveness fails (but the happy path should still return `Failed` if other validations fail). If it's the latter, we should return `Failed` in both cases.